Sources/Testing/Testing.docc/known-issues.md

```diff
@@ -10,13 +10,135 @@ See https://swift.org/LICENSE.txt for license information
+
+```swift
+@Test func grillHeating() throws {
+  var foodTruck = FoodTruck()
+  try foodTruck.startGrill()
+  #expect(foodTruck.grill.isHeating) // ❌ Expectation failed
+}
+```
+
+If the value of the `isHeating` property is `false`, `#expect` will record an
+issue. If you cannot fix the underlying problem, you can surround the failing
+code in a closure passed to `withKnownIssue()`:
+
+```swift
+@Test func grillHeating() throws {
+  var foodTruck = FoodTruck()
+  try foodTruck.startGrill()
+  withKnownIssue("Propane tank is empty") {
+    #expect(foodTruck.grill.isHeating) // Known issue
+  }
+}
+```
+
+The issue recorded by `#expect` will then be considered "known" and the test
+will not be marked as a failure. You may include an optional comment to explain
+the problem or provide context.
+
+### Mark a thrown error as known
+
+If an `Error` is caught by the closure passed to `withKnownIssue()`, the issue
+representing that caught error will be marked as known. Continuing the previous
+example, suppose the problem is that the `startGrill()` function is throwing an
+error. You can apply `withKnownIssue()` to this situation as well:
+
+```swift
+@Test func grillHeating() {
+  var foodTruck = FoodTruck()
+  withKnownIssue {
+    try foodTruck.startGrill() // Known issue
+    #expect(foodTruck.grill.isHeating)
+  }
+}
+```
+
+Because all errors thrown from the closure are caught and interpreted as known
+issues, the `withKnownIssue()` function is not throwing. Consequently, any
+subsequent code which depends on the throwing call having succeeded (such as the
+`#expect` after `startGrill()`) must be included in the closure to avoid
+additional issues.
+
+### Match a specific issue
+
+By default, `withKnownIssue()` considers all issues recorded while invoking the
+body closure known. If multiple issues may be recorded, you can pass a trailing
+closure labeled `matching:` which will be called once for each recorded issue
+to determine whether it should be treated as known:
+
+```swift
+@Test func batteryLevel() throws {
+  var foodTruck = FoodTruck()
+  try withKnownIssue {
+    let batteryLevel = try #require(foodTruck.batteryLevel) // Known
+    #expect(batteryLevel >= 0.8) // Not considered known
+  } matching: { issue in
+    guard case .expectationFailed(let expectation) = issue.kind else {
+      return false
+    }
+    return expectation.isRequired
+  }
+}
+```
+
+### Resolve a known issue
+
+If there are no issues recorded while calling `function`, `withKnownIssue()`
+will record a distinct issue about the lack of any issues having been recorded.
+This notifies you that the underlying problem may have been resolved so that you
+can investigate and consider removing `withKnownIssue()` if it's no longer
+necessary.
+
+### Handle a nondeterministic failure
+
+If `withKnownIssue()` sometimes succeeds but other times records an issue
+indicating there were no known issues, this may indicate a nondeterministic
+failure or a "flaky" test.
```

Comment:
Might want a paragraph here emphasizing "fix your tests first". We get a fair amount of feedback asking "why even _have_ XFAIL semantics?"

Reply:
I've added a paragraph about this now